staging/src/k8s.io/component-base/logs/validate.go

```diff
@@ -30,10 +26,10 @@ import (
 func ValidateLoggingConfiguration(c *config.LoggingConfiguration, fldPath *field.Path) field.ErrorList {
 	errs := field.ErrorList{}
 	if c.Format != DefaultLogFormat {
-		allFlags := UnsupportedLoggingFlags(hyphensToUnderscores)
```

Comment:
Why is  this changing behavior?

Reply:
Dropping the normalization isn't a change of behavior, the flags that `hyphensToUnderscore` was applied to already had underscores. To be honest, I don't know what it was supposed to achieve here.

Perhaps it was supposed to make the error message match the command's actual flags, but then this would have had to be `underscoresToHyphens` and even that is making assumptions about the command (not all use hyphens...).

Anyway, this is current master:
```
$ go run ./cmd/kube-apiserver --help | grep add-dir-header
                                            Non-default formats don't honor these flags: --add-dir-header, --alsologtostderr, --log-backtrace-at, --log-dir, --log-file, --log-file-max-size, --logtostderr, --one-output, --skip-headers, --skip-log-headers, --stderrthreshold, --vmodule, --log-flush-frequency.
      --add-dir-header                   If true, adds the file directory to the header of the log messages

$ go run ./cmd/kube-apiserver --logging-format=json --add-dir-header
...
Error: [... format: Invalid value: "json": Non-default format doesn't honor flag: add_dir_header]
exit status 1
```

Note how it uses `add_dir_header` here when the command line flag is `--add-dir-header`.

I touched this code because `flagIsSet` relied on having the flags added to the global `pflag.CommandLine`, which doesn't work anymore when not adding the flags there.

My change is problematic for a different reason as I found when checking this: because `UnsupportedLoggingFlags` always runs `AddGoFlagSet` anew, it doesn't properly capture the default value (it uses the *current* value as default, not the one from before parsing) and the "was set" check below doesn't work, i.e. this validation error didn't trigger anymore.

The fix is to capture the current state of the flags during init (but not in pflag.CommandLine!) and then use that copy during this check. I'll have that in my next revision.